pkg/modules/modules.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package modules
```

Comment:
Please move this into `pkg/datapath/linux` as we want to isolate all new Linux specific code

Reply:
1. This package handles only kernel modules listing and loading. I don't think that keeping it inside `pkg/datapath` package makes sense, because it does not relate to datapath itself.
2. What about moving all Linux-specific stuff to `modules_linux.go` and trying to keep `modules.go` OS-agnostic? I guess that anyone who wants to support the other platforms could just create i.e. a `modules_darwin.go` or `modules_freebsd.go` module.

Reply:
@tgraf ^

Reply:
I moved the `modprobe` usage to `modules_linux.go` and privileged unit test to `modules_linux_privileged_test` with a `linux` build constraint, so now that module should be independent from platform. Developing any other `modules_<platform>.go` module which implements `moduleLoader` function (returning the name of the tool, like i.e. `kldload` in FreeBSD) should be possible now, hopefully without touching the main module. Please let me know whether you agree with that solution.